dotnet/src/Connectors/VectorData.Abstractions/ConnectorSupport/VectorStoreRecordModelBuildingOptions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.VectorData.ConnectorSupport;
+
+/// <summary>
+/// Contains options affecting model building; passed to <see cref="VectorStoreRecordModelBuilder"/>.
+/// </summary>
+[Experimental("MEVD9001")]
+public sealed class VectorStoreRecordModelBuildingOptions
+{
+    /// <summary>
+    /// Whether multiple key properties are supported.
+    /// </summary>
+    public required bool SupportsMultipleKeys { get; set; }
```

Comment:
I like the fact that you are using `required` to make it enforced by the compiler 👍